content/chapters/app-interact/lab/content/os-cloud.md

```diff
@@ -3,6 +3,27 @@
 In this section we are going to build a "toy cloud" called `OS Cloud`.
 Similar to a real cloud (like `aws`), `OS Cloud` will allow us to create and manage virtual machines, through an `http` API.
 
+## Containers vs VMs
+
+Containers are a lightweight virtualization technology that allows multiple isolated user-space instances to run on a single host operating system.
+They are often compared to chroots because they both provide isolated environments for running applications.
+However, containers take this isolation a step further by using kernel features such as namespaces and cgroups to provide a more complete and secure isolation of resources.
```

Comment:
Go into a bit of detail regarding cgroups and namespaces. You can draw inspiration from these resources and place them in the text as further reading:
- https://www.nginx.com/blog/what-are-namespaces-cgroups-how-do-they-work/
- https://www.baeldung.com/linux/cgroups-and-namespaces

Reply:
This is the shit for `namespaces`:
- https://blog.quarkslab.com/digging-into-linux-namespaces-part-1.html